File: fractal_flutter_test_bk/src/main/java/com/azpdev/fractal_flutter_test_bk/model/OrderDetail.java
package com.azpdev.fractal_flutter_test_bk.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;

@Entity
public class OrderDetail {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    Integer id;

    @Column(name = "order_id")
    private Integer orderId;

    @Column(name = "product_id")
    private Integer productId;
    Double price;
    Integer qty;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Integer getOrder_id() {
        return orderId;
    }

    public void setOrder_id(Integer order_id) {
        this.orderId = order_id;
    }

    public Integer getProduct_id() {
        return productId;
    }

    public void setProduct_id(Integer product_id) {
        this.productId = product_id;
    }

    public Double getPrice() {
        return price;
    }

    public void setPrice(Double price) {
        this.price = price;
    }

    public Integer getQty() {
        return qty;
    }

    public void setQty(Integer qty) {
        this.qty = qty;
    }

}

File: fractal_flutter_test/lib/model/OrderDetail.dart
class OrderDetail {
  final int orderId;
  final int productId;
  final double price;
  final int qty;

  OrderDetail({
    required this.orderId,
    required this.productId,
    required this.price,
    required this.qty,
  });

  Map<String, dynamic> toJson() {
    return {
      'order_id': orderId,
      'product_id': productId,
      'price': price,
      'qty': qty,
    };
  }

  factory OrderDetail.fromJson(Map<String, dynamic> json) {
    return OrderDetail(
      orderId: json['order_id'],
      productId: json['product_id'],
      price: json['price'],
      qty: json['qty'],
    );
  }
}